llmfoundry/utils/builders.py

```diff
@@ -208,14 +210,16 @@ def build_tokenizer(
             int(1e30),
         )
 
-    if dist.get_local_rank() == 0:
-        with open(signal_file_path, 'wb') as f:
-            f.write(b'local_rank0_completed_tokenizer_setup')
+    if dist.is_available() and dist.is_initialized(
+    ) and dist.get_world_size() > 1:
+        if dist.get_local_rank() == 0:
+            with open(signal_file_path, 'wb') as f:
+                f.write(b'local_rank0_completed_tokenizer_setup')
 
-    dist.barrier()
+        dist.barrier()
 
-    if dist.get_local_rank() == 0:
-        os.remove(signal_file_path)
+        if dist.get_local_rank() == 0:
```

Comment:
Does this need a `if dist.is_available() and dist.is_initialized() and and dist.get_world_size() > 1`? 

Might be useful to wrap in helper function as well

Reply:
its not used much...almost everything is run after initialize dist is called and we don't actually need to check this. Also the line you commented on is inside the above available/initialized check, yeah? If I'm misreading, that would def be a bug

Reply:
oh yeah nice I misread thanks!